src/main/java/org/distributed/systems/chord/actors/Node.java

```diff
@@ -143,54 +148,97 @@ public Receive createReceive() {
-                        System.out.println("Node's " + this.id + "current predecessor: " + this.predecessorId);
-                        System.out.println("Node's " + this.id + " new successor: " + msg.newSucessorKey);
-                    } else if (msg.newPredecessor != null && msg.newSucessor == null) {
-                        // it's a predecessor
-                        this.predecessorId = msg.newPredecessorKey;
-                        this.predecessor = msg.newPredecessor;
-                        JoinMessage.JoinConfirmationReply confirmReplyMsg = new JoinMessage.JoinConfirmationReply(true);
-                        getContext().getSender().tell(confirmReplyMsg, ActorRef.noSender());
-                        System.out.println("Node " + this.id + " confirmed the JoinRequest");
-                        System.out.println("Node's " + this.id + "current successor: " + this.sucessorId);
-                        System.out.println("Node's " + this.id + " new predecessor: " + msg.newPredecessorKey);
+                .match(FindSuccessor.Request.class, msg -> {
+                    // +1 to do inclusive interval
+                    // Single Node Edge Case: this.id == this.succId
+                    if (this.id == this.fingerTableSuccessor().id) {
+                        getContext().getSender().tell(new FindSuccessor.Reply(this.fingerTableSuccessor().chordRef, this.fingerTableSuccessor().id, msg.fingerTableIndex), getSelf());
+                    } else if (isBetweenExeclusive(this.id, this.fingerTableSuccessor().id + 1, msg.id)) {
+                        getContext().getSender().tell(new FindSuccessor.Reply(this.fingerTableSuccessor().chordRef, this.fingerTableSuccessor().id, msg.fingerTableIndex), getSelf());
                     } else {
-                        // Some illegal state, decline this request
-                        System.out.println("Node " + this.id + " DECLINED the JoinRequest");
-                        getContext().getSender().tell(new JoinMessage.JoinConfirmationReply(false), ActorRef.noSender());
+                        // this.fingerTableSuccessor().chordRef.forward(msg, getContext());
+                        ActorRef ndash = this.closest_preceding_node((int) msg.id);
+                        ndash.forward(msg, getContext());
                     }
                 })
+                .match(FixFingers.Request.class, msg -> {
+                    this.fix_fingers();
+                })
+                .match(UpdateFinger.Request.class, msg -> {
+                    StringBuilder sb = new StringBuilder();
+                    sb.append("Updating Finger Table Entry Number " + msg.fingerTableIndex + "\n");
+                    sb.append("\t Previous Entry:\t " + this.fingerTable[(int) msg.fingerTableIndex] + "\n");
+                    sb.append("\t New Entry:\t " + msg.fingerTableEntry + "\n");
+                    this.fingerTable[(int) msg.fingerTableIndex] = msg.fingerTableEntry;
```

Comment:
fingerTableIndex in message should just be int I guess. Don't think we will have values greater than 160 for that anyway ;p

Reply:
:+1: Your are right, think I did not think about that one!